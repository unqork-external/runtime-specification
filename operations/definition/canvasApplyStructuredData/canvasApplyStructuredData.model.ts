import { DiscriminatorValue, Required } from '@tsed/schema'

import { CanvasApplyStructuredDataOperationOptions } from './canvasApplyStructuredData.options'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@DiscriminatorValue(OperationTypes.CANVAS_APPLY_STRUCTURED_DATA)
export class CanvasApplyStructuredDataOperation extends Operation<OperationTypes.CANVAS_APPLY_STRUCTURED_DATA> {
  @Required()
  public options: CanvasApplyStructuredDataOperationOptions = new CanvasApplyStructuredDataOperationOptions()
}
