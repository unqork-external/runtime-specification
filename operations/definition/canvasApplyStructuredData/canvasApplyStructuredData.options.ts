import { Default, Description, Example, Optional } from '@tsed/schema'

import { TargetedOperationOptions } from '../../interface'

export class CanvasApplyStructuredDataOperationOptions extends TargetedOperationOptions {
  @Optional()
  @Description('the id of layout you want to start collecting from')
  @Example('root')
  @Default('root')
  layoutId?: string

  @Optional()
  @Description('Use to get structured data and override default exclude functionality')
  withExcludes?: boolean

  @Optional()
  @Description('Normalizes all childLayouts to a single array under key "children"')
  normalizeChildren?: boolean

  @Optional()
  @Description('Use to get structured data but only with the values from the desired paths per component')
  @Example('["key"]')
  paths?: string[]
}
