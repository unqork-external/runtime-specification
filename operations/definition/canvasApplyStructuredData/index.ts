export * from './canvasApplyStructuredData.model'
export * from './canvasApplyStructuredData.options'
