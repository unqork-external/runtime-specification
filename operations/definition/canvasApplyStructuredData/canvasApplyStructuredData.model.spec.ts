import { CanvasApplyStructuredDataOperation } from './canvasApplyStructuredData.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('CanvasApplyStructuredDataOperation Model', () => {
  let validate, schema
  beforeAll(() => {
    const response = generateSchemaAndValidate(CanvasApplyStructuredDataOperation)
    validate = response.validate
    schema = response.schema
  })

  it('should match this JSON Schema', () => {
    expect(schema).toMatchSnapshot()
  })
})
