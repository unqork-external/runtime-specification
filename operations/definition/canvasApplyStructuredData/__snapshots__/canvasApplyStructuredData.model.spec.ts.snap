// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CanvasApplyStructuredDataOperation Model should match this JSON Schema 1`] = `
{
  "definitions": {
    "CanvasApplyStructuredDataOperationOptions": {
      "properties": {
        "inputs": {
          "description": "
    Inputs to be used within formulas in other operation options
  ",
          "items": {
            "$ref": "#/definitions/InputRef",
          },
          "type": "array",
        },
        "layoutId": {
          "default": "root",
          "description": "the id of layout you want to start collecting from",
          "examples": [
            "root",
          ],
          "type": "string",
        },
        "normalizeChildren": {
          "description": "Normalizes all childLayouts to a single array under key "children"",
          "type": "boolean",
        },
        "paths": {
          "description": "Use to get structured data but only with the values from the desired paths per component",
          "examples": [
            "["key"]",
          ],
          "type": "array",
        },
        "shouldExecute": {
          "description": "
    Conditionally execute this operation based on a formula.
    Can set to false to temporarily disabled when debugging
  ",
          "type": [
            "string",
            "boolean",
          ],
        },
        "targetKey": {
          "description": "Key or path that this operation should target or operate upon",
          "examples": [
            "myComponent",
            "panel.*",
            "grid.row(#).col(name)",
          ],
          "minLength": 1,
          "type": "string",
        },
        "withExcludes": {
          "description": "Use to get structured data and override default exclude functionality",
          "type": "boolean",
        },
      },
      "required": [
        "targetKey",
      ],
      "type": "object",
    },
    "InputRef": {
      "description": "Inputs can be used to utlize dynamic values from other state within the runtime
  within a particular execution of an operation",
      "properties": {
        "alias": {
          "description": "Optionally alias resolved value for use in formulas",
          "type": "string",
        },
        "propertyPath": {
          "description": "Path to property on target to resolve",
          "type": "string",
        },
        "required": {
          "description": "If true, operation won't execute if targeted property has no value",
          "type": "boolean",
        },
        "silent": {
          "description": "Not used in later formulas, only used to determine if operation should execute",
          "type": "boolean",
        },
        "targetKey": {
          "description": "Key of target this input depends on",
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "targetKey",
      ],
      "type": "object",
    },
  },
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/CanvasApplyStructuredDataOperationOptions",
    },
    "type": {
      "const": "CANVAS_APPLY_STRUCTURED_DATA",
      "examples": [
        "CANVAS_APPLY_STRUCTURED_DATA",
      ],
      "type": "string",
    },
  },
  "required": [
    "options",
  ],
  "type": "object",
}
`;
